### DISCUSSION TITLE
Convert to Managed parameter in PAC

### DISCUSSION THREAD

**Discussion by sasidamodharan** [microsoft/powerplatform-build-tools#968] [General]
Hi, We have recently starting using PAC command to import solution. Earlier, we were using Import-SolutionAsync to import the solution in our pipeline, which suddenly starting failing about 2 weeks back.
My question is regarding the parameter in PAC "Convert to Managed" - Has anyone used this? Does this covert any unmanaged layer in the imported components to managed automatically? We tried this parameter, but the unmanaged layer remained in the system.
Any thoughts around this?

**Comment by parvezghumra** [comment]
@sasidamodharan The 'Convert to Managed' parameter is designed to convert to managed any unmanaged components found in the target environment where a managed version is included in the solution being imported. This parameter does not remove the active customisations in the unmanaged layer.

**Reply by jbuenoliveira** [reply]
@sasidamodharan - suggestion: then you could try the `--force-overwrite true`, it forces an overwrite of unmanaged customizations in the target environment.

As an alternative, you could achieve this via PowerShell by deleting any active customizations from a specific component. e.g.:

```PowerShell
# it requires 'Microsoft.Xrm.Data.PowerShell' module
$request = New-Object 'Microsoft.Xrm.Sdk.OrganizationRequest' -ArgumentList 'RemoveActiveCustomizations'
# component Types: https://learn.microsoft.com/en-us/power-apps/developer/data-platform/webapi/reference/solutioncomponent?view=dataverse-latest
$request.Parameters.Add('SolutionComponentName', "$ComponentType")
$request.Parameters.Add('ComponentId', [Guid]$ComponentId)
Invoke-Command -ScriptBlock { $CrmConn.Execute($request) } | Out-Null
```

**Comment by SabrinaDiBartolomeo** [comment]
The convert-to-managed flag is deprecated and is no longer used by the platform. When a managed solution is imported, the platform will automatically convert any unmanaged components in the target environment that are included in the solution to managed, no matter the flag's setting.

**Reply by petrochuk** [reply]
Next version of PAC CLI will ignore convert-to-managed argument
